### PR TITLE
Seek redesign: add implementation of Set source position

### PR DESCRIPTION
**Pull request by skywojciechowskim**
Summary: Remake Seek procedure. New method - setSourcePosition added.
Type: Feature
Test Plan: Unittests, Component Tests, YT Conformance Tests, NPLB
Jira: RIALTO-491

### CODE REVIEW

File: media/server/gstplayer/source/tasks/generic/SetSourcePosition.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+    if (!source)
+    {
+        RIALTO_SERVER_LOG_WARN("failed to set source position - source is NULL");
+        return;
+    }
+
+    GstSegment *segment{m_gstWrapper->gstSegmentNew()};
+    m_gstWrapper->gstSegmentInit(segment, GST_FORMAT_TIME);
+    m_gstWrapper->gstSegmentDoSeek(segment, m_context.playbackRate, GST_FORMAT_TIME, GST_SEEK_FLAG_NONE,
```
[the remark this answers is not in the document]
Comment (skywojciechowskim) [reply media/server/gstplayer/source/tasks/generic/SetSourcePosition.cpp]: check added.

File: media/server/gstplayer/source/tasks/generic/SetSourcePosition.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    GstElement *source{nullptr};
+    auto sourceElem = m_context.streamInfo.find(m_type);
+    if (sourceElem != m_context.streamInfo.end())
+    {
+        source = sourceElem->second.appSrc;
+    }
+    if (!source)
+    {
+        RIALTO_SERVER_LOG_WARN("failed to set source position - source is NULL");
+        return;
```
[the remark this answers is not in the document]
Comment (skywojciechowskim) [reply media/server/gstplayer/source/tasks/generic/SetSourcePosition.cpp:59]: Added playback error (failure), the same behavior as in "old" SetPosition

File: tests/componenttests/client/tests/mse/SetSourcePositionTest.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+ *   Server notifies the client that set source position procedure has been queued in server
+ *   Expect that the procedure status is propagated to the client.
+ *
+ * Test Teardown:
+ *  Terminate the media session.
+ *  Memory region created for the shared buffer is closed.
+ *  Server is terminated.
+ *
+ * Expected Results:
+ *  Set Source Position is handled and forwarded to the server for PLAYING, PAUSED and END_OF_STREAM state.
```
[the remark this answers is not in the document]
Comment (skywojciechowskim) [reply tests/componenttests/client/tests/mse/SetSourcePositionTest.cpp]: description changed.

File: tests/componenttests/server/tests/mediaPipeline/SetSourcePositionTest.cpp [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+ * Test Teardown:
+ *  Memory region created for the shared buffer is unmapped.
+ *  Server is terminated.
+ *
+ * Expected Results:
+ *  Set Source Position is handled and forwarded to the gstreamer.
+ *
+ * Code:
+ */
+TEST_F(SetSourcePositionTest, setAudioSourcePositionSuccess)
```
[the remark this answers is not in the document]
Comment (skywojciechowskim) [reply tests/componenttests/server/tests/mediaPipeline/SetSourcePositionTest.cpp:179]: CT modified as well.

File: tests/componenttests/server/tests/mediaPipeline/SetSourcePositionTest.cpp [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
```
Comment (skywilliamsonl) [inline tests/componenttests/server/tests/mediaPipeline/SetSourcePositionTest.cpp:179]: Change name to just setSourcePositionSuccess

**Review by skywilliamsonl — DISMISSED** [review]

File: tests/componenttests/server/tests/mediaPipeline/SetSourcePositionTest.cpp (hunk 5 above)
Comment (skywojciechowskim) [reply tests/componenttests/server/tests/mediaPipeline/SetSourcePositionTest.cpp]: changed.

File: media/server/gstplayer/source/GstGenericPlayer.cpp [hunk 6]
```diff
@@ -630,6 +648,41 @@ bool GstGenericPlayer::setCodecData(GstCaps *caps, const std::shared_ptr<CodecDa
...
+                                        GST_SEEK_TYPE_SET, initialPosition->second, GST_SEEK_TYPE_SET,
+                                        GST_CLOCK_TIME_NONE, nullptr))
+    {
+        RIALTO_SERVER_LOG_WARN("Segment seek failed.");
+        m_gstWrapper->gstSegmentFree(segment);
+        m_context.initialPositions.erase(initialPosition);
+        return false;
+    }
+
+    RIALTO_SERVER_LOG_INFO("New segment: [%" GST_TIME_FORMAT ", %" GST_TIME_FORMAT "], rate: %f \n",
```
Comment (aczs) [inline media/server/gstplayer/source/GstGenericPlayer.cpp:671]: Maybe change to milestone. It's worth to have important seek logs as milestone. They shouldn't flood us and might be helpful.

**Review by aczs — APPROVED** [review]

File: media/server/gstplayer/source/GstGenericPlayer.cpp (hunk 6 above)
Comment (skywojciechowskim) [reply media/server/gstplayer/source/GstGenericPlayer.cpp]: severity changed

**Review by aczs — APPROVED** [review]

**Pull request merged**